tests/files/test-fix-block-img-tag-sizes.php

```diff
@@ -0,0 +1,110 @@
+			),
+			'no-html-api'       => array(
+				'is_admin'      => false,
+				'block_content' => '<figure class="wp-block-image size-medium"><img src="test.jpg" class="wp-image-123" /></figure>',
+				'expected'      => '<figure class="wp-block-image size-medium"><img src="test.jpg" class="wp-image-123" /></figure>',
+			),
+			'no-img-tag'        => array(
+				'is_admin'      => false,
+				'block_content' => '<figure class="wp-block-image size-medium">No image here</figure>',
+				'expected'      => '<figure class="wp-block-image size-medium">No image here</figure>',
+			),
+			'no-image-id'       => array(
+				'is_admin'      => false,
+				'block_content' => '<figure class="wp-block-image size-medium"><img src="test.jpg" class="no-image-id" /></figure>',
+				'expected'      => '<figure class="wp-block-image size-medium"><img src="test.jpg" class="no-image-id" /></figure>',
+			),
+			'no-figure-tag'     => array(
+				'is_admin'      => false,
+				'block_content' => '<img src="test.jpg" class="wp-image-123" />',
+				'expected'      => '<img src="test.jpg" class="wp-image-123" />',
+			),
+			'no-size-class'     => array(
+				'is_admin'      => false,
+				'block_content' => '<figure class="wp-block-image"><img src="test.jpg" class="wp-image-123" /></figure>',
+				'expected'      => '<figure class="wp-block-image"><img src="test.jpg" class="wp-image-123" /></figure>',
+			),
+			'full-size'         => array(
+				'is_admin'      => false,
+				'block_content' => '<figure class="wp-block-image size-full"><img src="test.jpg" class="wp-image-123" /></figure>',
+				'expected'      => '<figure class="wp-block-image size-full"><img src="test.jpg" class="wp-image-123" /></figure>',
+			),
+			'successful-update' => array(
+				'is_admin'      => false,
+				'block_content' => '<figure class="wp-block-image size-medium"><img src="test.jpg" class="wp-image-123" /></figure>',
+				// THIS IS FLAKY - the actual output may vary depending on the order of the attributes passed, but it seems that the last passed arguments get at the beginning of the tag
+				'expected'      => '<figure class="wp-block-image size-medium"><img height="200" width="300" src="test.jpg" class="wp-image-123" /></figure>',
+				'metadata'      => array(
+					'sizes' => array(
+						'medium' => array(
+							'width'  => 300,
+							'height' => 200,
+						),
+					),
+				),
+			),
+		);
+	}
+
+	/**
+	 * Test the fix_img_block_sizes method.
+	 *
+	 * @dataProvider get_data_for_fix_img_block_sizes
+	 * @param bool   $is_admin      Whether we're in admin context.
+	 * @param string $block_content The block content to process.
+	 * @param string $expected      The expected output.
+	 * @param array  $metadata      Optional metadata for the image.
+	 * @return void
+	 */
+	public function test__fix_img_block_sizes( $is_admin, $block_content, $expected, $metadata = null ) {
+		
+		if ( $is_admin && ! defined( 'WP_ADMIN' ) ) {
+			define( 'WP_ADMIN', true );
+		}
+
```

Comment:
I don't like it because it might affect other tests.

I suggest moving the `is_admin()` check around the `add_filter()` call.